.github/workflows/docs.yml

```diff
@@ -27,30 +22,116 @@ jobs:
     # Only run in original repo (not in forks)
     if: github.repository == 'django-components/django-components'
     steps:
+
+      ##############################
+      # SETUP
+      ##############################
+
       - name: Checkout
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
+
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
-          python-version: "3.12"
+          python-version: "3.13"
+          cache: 'pip'
 
       - name: Install Hatch
         run: |
             python -m pip install --upgrade pip wheel
-            python -m pip install -q hatch pre-commit
+            python -m pip install -q hatch pre-commit asv
             hatch --version
 
-      - name: Create Virtual Environment
-        run: hatch env create docs
-
       - name: Configure git
         run: |
+          # Get the master branch so we can run benchmarks on it
+          git remote add upstream https://github.com/${{ github.repository }}.git
+          git fetch origin master:master
+          git checkout master
+
           # required for "mike deploy" command below which pushes to gh-pages
           git config user.name github-actions
           git config user.email github-actions@github.com
 
+      ###########################################
+      # RECORD BENCHMARK - ONLY ON PUSH TO MASTER
+      ###########################################
+
+      - name: Run benchmarks for tag
```

Comment:
This is the workflow to generate benchmark data on release (tag push), and include the data as part of the documentation site.

This is similar to the workflow in `benchmark.yml` (comparison on PR). Except here:
1. Instead of comparing 2 commits, we just run benchmarks for a single commit - the latest commit on master.
2. We then run `asv publish` to generate a directory with the web report
3. We push the benchmarking results AND the web report to master.

So this way, we actually "save" the results for tags, and make them available via the documnetaiton site.